api/types.go

```diff
@@ -0,0 +1,47 @@
+package api
+
+type Context struct {
+	State DebuggerState `json:"state"`
```

Comment:
I am all for expanding DebuggerState to include more relevant information (current pc, current thread, current func, current go routine, etc..)

To make this a bit more concrete, what would be a case for when we would need access to both sides of the coin simultaneously? Also, when would this DebuggerState type come into play, as the result of any state changing call (continue, step, next, etc...)?

I feel that, if your on the receiving end of the DebuggerState, you likely know what action you've initiated, and if you're the debug server you already know the state, you're just packaging it for client consumption.

I may be missing some of the benefits of this pattern however.